src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -159,6 +164,12 @@ private[health] class HealthCheckActor(
     }
   }
 
+  def checkEnoughInstancesRunning(): Boolean = {
+    val activeInstances = Await.result(instanceTracker.countActiveSpecInstances(app.id), 5 seconds)
+    val futureHealthyCapacity: Double = (activeInstances - 1) / app.instances.toDouble
+    futureHealthyCapacity >= app.upgradeStrategy.minimumHealthCapacity
```

Comment:
Since the patch seems pretty minimal, why not taking the opportunity of adding a dedicated field in the UpgradeStrategy class ?